ansible_base/authentication/authenticator_plugins/ldap.py

```diff
@@ -69,8 +72,71 @@ def validator(value):
 
 
 class LDAPSearchField(ListField):
+    @staticmethod
+    def is_single_search(value):
+        try:
+            # Note: this method was chosen as it was how AWX determined a single vs multi-field
+            return len(value) == 3 and type(value[0]) is str
+        except TypeError:
+            return False
+
+    def validate_single_search(self, search_term):
+        errors = {}
+
+        # If this thing doesn't look like we expect, return an error
+        if not LDAPSearchField.is_single_search(search_term):
+            return _MUST_BE_AN_ARRAY_MESSAGE_TRANSLATED
```

Comment:
Return as a dict to match errors? (or set into errors and return directly?)  The comparison later for "no error" is comparing to {}, which would work fine but is a bit odd since the return can be a string or dict.

Reply:
Yeah, no problem as is..